### PR TITLE
Regd-81-remove-overlapping-hits

### PR DESCRIPTION
**Pull request by shengchd**
- filtered overlapping peaks from the same datasets in ChIP, DNase & ATAC-seq
- corrected the total peak number after filtering

### CODE REVIEW

File: src/encoded/static/components/regulome_search.js [hunk 1]
```diff
@@ -544,6 +544,32 @@ const populationOrder = [
...
 
+// function to de-dup overlapping peaks in each dataset from ChIP, DNase and ATAC-seq assays
+function filter_overlapping_peaks(Datasets) {
+    var Datasets_filtered = [];
+    var Datasets_filtered_count = 0;
+    if (Datasets.length > 0) {
+        //we want to keep the wider peaks in each dataset
+        //if the start & end positions are the same we will keep the one with the strongest signal
+        //sort all peaks by dataset ids -> start positions in ascending order -> end positions in descending order -> signals in descending order
+        Datasets.sort((a,b)=>a.dataset_rel.localeCompare(b.dataset_rel) || a.start-b.start || b.end-a.end || b.value-a.value);
```
Comment (mingjiecn) [inline src/encoded/static/components/regulome_search.js]: value is not used for sorting later, do we still need to sort by value?

File: src/encoded/static/components/regulome_search.js [hunk 2]
```diff
@@ -544,6 +544,32 @@ const populationOrder = [
...
+            if (Datasets[i].dataset_rel !== lastDataset || Datasets[i].end > lastEnd) { 
+                Datasets_filtered.push(Datasets[i]); 
+                lastDataset = Datasets[i].dataset_rel;
+                lastEnd = Datasets[i].end;
+            } else {
+                Datasets_filtered_count ++;
+            }
+        }
+    }
+    return [Datasets_filtered,Datasets_filtered_count];
```
Comment (mingjiecn) [inline src/encoded/static/components/regulome_search.js]: Should have one space after comma. Did you lint your code before pushing?

**Review by mingjiecn — CHANGES_REQUESTED** [review]
some minor concerns need to be addressed

File: src/encoded/static/components/regulome_search.js (hunk 1 above)
Comment (shengchd) [reply src/encoded/static/components/regulome_search.js]: We are showing signals (i.e. values) for ChIP-seq peaks, so I thought it's better to explicitly keep the one with the strongest signal this way (when there are multiple peaks with exactly same positions from the same dataset)?

File: src/encoded/static/components/regulome_search.js (hunk 2 above)
Comment (shengchd) [reply src/encoded/static/components/regulome_search.js]: Thanks for pointing out that, my eslint wasn't working correctly. I have also fixed other stylistic issues in my code.

File: src/encoded/static/components/regulome_search.js (hunk 1 above)
Comment (mingjiecn) [reply src/encoded/static/components/regulome_search.js]: sound good!

**Review by mingjiecn — APPROVED** [review]

**Pull request merged**